src/TiledArray/dist_array.h

```diff
@@ -407,48 +422,53 @@ class DistArray : public madness::archive::ParallelSerializableObject {
   /// Global object id
 
   /// \return A globally unique identifier.
+  /// \throw TiledArray::Exception if the PIMPL has not been initialized.
   /// \note This function is primarily used for debugging purposes. Users
   /// should not rely on this function.
-  madness::uniqueidT id() const { return pimpl_->id(); }
+  madness::uniqueidT id() const { return check_pimpl().id(); }
```

Comment:
better name for `check_pimpl()` is `checked_pimpl()` or even `pimpl()`. Principle of least surprise dictates that `check_pimpl()` performs action and does not return a value.